### PR TITLE
fix(ui): Multiple UI fixes

### LINKED ISSUE #5496: Label choices aren't remembered
## Summary 

1. Go to /workflows and choose a valid label and namespace. Filter your results by this label.
2. Click onto the archived workflows tab
3. Click back to /workflows

**issue:**
The namespace is retained, but the selected label is not.

This also occurs on the /archived-workflows view.

## Diagnostics

What Kubernetes provider are you using? EKS 1.19

What version of Argo Workflows are you running? 3.0 rc8

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by simster7**
Fixes #5496 
Fixes https://github.com/argoproj/argo-workflows/issues/5494

### CODE REVIEW

File: ui/src/app/workflows/components/workflows-list/workflows-list.tsx [hunk 1]
```diff
@@ -95,16 +97,19 @@ export class WorkflowsList extends BasePage<RouteComponentProps<any>, State> {
...
         this.state = {
             pagination: {
                 offset: this.queryParam('offset'),
                 limit: parseLimit(this.queryParam('limit')) || savedOptions.paginationLimit || 500
             },
             namespace: this.props.match.params.namespace || '',
-            selectedPhases: this.queryParams('phase').length > 0 ? (this.queryParams('phase') as WorkflowPhase[]) : savedOptions.selectedPhases,
-            selectedLabels: this.queryParams('label'),
+            selectedPhases: phaseQueryParam.length > 0 ? (phaseQueryParam as WorkflowPhase[]) : savedOptions.selectedPhases,
+            selectedLabels: labelQueryParam.length > 0 ? (labelQueryParam as string[]) : savedOptions.selectedLabels,
```
Comment (simster7) [inline ui/src/app/workflows/components/workflows-list/workflows-list.tsx:112]: Fixes https://github.com/argoproj/argo-workflows/issues/5496

File: ui/src/app/shared/components/object-editor/object-editor.tsx [hunk 2]
```diff
@@ -91,7 +91,12 @@ export const ObjectEditor = <T extends any>({type, value, buttons, onChange}: Pr
                         renderIndentGuides: false,
                         scrollBeyondLastLine: true
                     }}
-                    onChange={() => setIsModified(true)}
+                    onChange={v => {
+                        if (!isModified) {
+                            setText(v);
+                            setIsModified(true);
+                        }
+                    }}
```
Comment (simster7) [inline ui/src/app/shared/components/object-editor/object-editor.tsx:99]: Fixes https://github.com/argoproj/argo-workflows/issues/5493

Comment (alexec) [reply ui/src/app/shared/components/object-editor/object-editor.tsx:99]: I have a fix in another PR for this, I think it needs to be a bit more involved. Can you revert this one change please?

**Review by alexec — APPROVED** [review]

File: ui/src/app/shared/components/object-editor/object-editor.tsx (hunk 2 above)
Comment (simster7) [reply ui/src/app/shared/components/object-editor/object-editor.tsx:99]: Absolutely

File: ui/src/app/login/components/login.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const getRedirect = (): string => {
     if (urlParams.has('redirect')) {
         return 'redirect=' + urlParams.get('redirect');
     }
-    return '';
+    return 'redirect=' + window.location.origin + '/workflows';
```
Comment (simster7) [inline ui/src/app/login/components/login.tsx:21]: Fixes https://github.com/argoproj/argo-workflows/issues/5494

**Review by alexec — APPROVED** [review]

**Pull request merged**